File: app/src/main/java/com/devidea/timeleft/AppDatabase.java
package com.devidea.timeleft;

import androidx.room.Database;
import androidx.room.RoomDatabase;

@Database(entities = {EntityWidgetInfo.class, EntityItemInfo.class}, version = 2, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    public abstract DatabaseDao DatabaseDao();
}

File: app/src/main/java/com/devidea/timeleft/AppDatabase.kt
package com.devidea.timeleft

import androidx.room.Room
import androidx.room.Database
import androidx.room.RoomDatabase
import android.content.Context

@Database(
    entities = [EntityWidgetInfo::class, EntityItemInfo::class],
    version = 3,
    exportSchema = false
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun DatabaseDao(): DatabaseDao

    companion object {
        private var instance: AppDatabase? = null

        @Synchronized
        fun getInstance(context: Context): AppDatabase? {
            if (instance == null){
                synchronized(AppDatabase::class){
                    instance = Room.databaseBuilder(
                            context.applicationContext,
                            AppDatabase::class.java,
                            "appdatabase"
                    ).allowMainThreadQueries()
                .build()
                }
            }
            return instance
        }

    }
}